src/test/ui/hrtb/issue-58451.stderr

```diff
@@ -2,15 +2,19 @@ error[E0061]: this function takes 1 argument but 0 arguments were supplied
   --> $DIR/issue-58451.rs:12:9
    |
 LL |     f(&[f()]);
-   |         ^-- supplied 0 arguments
+   |         ^--
    |         |
-   |         expected 1 argument
+   |         an argument is missing
    |
 note: function defined here
   --> $DIR/issue-58451.rs:5:4
    |
 LL | fn f<I>(i: I)
-   |    ^    ----
+   |    ^
+help: provide the argument
+   |
+LL |     f(&[f({_})]);
+   |         ~~~~~~
```

Comment:
I'm worried that the `{_}` placeholder might be a bit abstract, but we can iterate on this after landing the PR.